core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -52,7 +52,7 @@ sealed trait Chunk[@specialized +A] { self =>
     val len = self.length
 
     if (n <= 0) self
-    else if (n == len) Chunk.empty
+    else if (n >= len) Chunk.empty
```

Comment:
Interesting that this did not show up in the property checks either. Good catch either way!